raft/src/main/java/org/apache/kafka/raft/QuorumState.java

```diff
@@ -505,7 +505,6 @@ public void prospectiveAddVotedState(
             );
         }
 
-        ProspectiveState prospectiveState = prospectiveStateOrThrow();
         // Note that we reset the election timeout after voting for a candidate because we
         // know that the candidate has at least as good of a chance of getting elected as us
```

Comment:
Outside the scope of this PR but this is not true when there is a network partition. We should not reset the election timer if a vote is granted. The worst case, if the replica gets this wrong, is that the replica transitions back to the follower or unattached state.

Reply:
hm, right, we no longer need to reset election timeouts because loss of prospective elections are non-disruptive

Reply:
https://issues.apache.org/jira/browse/KAFKA-19319
would be a good task for a raft newbie